modules/services/osm.js

```diff
@@ -987,8 +987,9 @@ export default {
                 .zoomExtent([_tileZoom, _tileZoom])
                 .getTiles(projection)
                 .map(function(tile) {
-                    tile.mapId = tile.id = layer.id;
+                    tile.mapId = layer.id;
                     tile.layerName = layer.name;
+                    tile.id = tile.id + '_' + tile.mapId;
```

Comment:
This was the crux of the bug.  The tile id is used as the key in the tile cache.  The old code just used the map id which obviously will be one entry per dataset layer.  We need the key to be a combo of the tile id and the map id.

A lot of the other code I removed was a workaround for this bug.

Reply:
And reverted to restore upstream iD lines of code